### PR TITLE
Update webpack to v4

### PR DESCRIPTION
**Pull request by fisker**
Update webpack to v4

### CODE REVIEW

File: scripts/build/bundler.js [hunk 1]
```diff
@@ -176,13 +176,14 @@ function getWebpackConfig(bundle) {
...
+      globalObject: 'new Function("return this")()'
     },
-    plugins: [
-      new webpack.DefinePlugin({
-        "process.env.NODE_ENV": JSON.stringify("production")
-      })
-    ]
+    optimization: {
+      // disable terser for `parser-postcss.js`
+      minimize: false
```
Comment (evilebottnawi) [inline scripts/build/bundler.js]: Why we disable this?

Comment (fisker) [reply scripts/build/bundler.js]: this module can't be minified, unknown reason, we can figure later

https://github.com/prettier/prettier/pull/6200#issuecomment-500018106

Comment (fisker) [reply scripts/build/bundler.js]: may be I should add a comment to this link or a TODO

Comment (evilebottnawi) [reply scripts/build/bundler.js]: @fisker yes, please leave comment in source code `// Todo description`

**Comment by fisker** [comment]
I just resolve the conflicts, anything else need to be done?

**Comment by lipis** [comment]
Is this good to go?

**Comment by evilebottnawi** [comment]
I think all good and we can safely merge

**Comment by evilebottnawi** [comment]
hm, minimizer: false output big js file, so it can slow site

**Comment by fisker** [comment]
@lipis @evilebottnawi this minimizer: false only effect `parser-postcss.js` (it's not also minified before this PR), see comment above. and I already made a seperate PR to fix it, https://github.com/prettier/prettier/pull/6395 so  it's good to go

**Pull request merged**